### ISSUE TITLE
Markdown roundtrip

### ISSUE THREAD

**Issue by remram44** [remram44/matrix-appservice-gitter-twisted#7]
Markdown posted on Matrix gets posted on Gitter fine, but markdown in Gitter shows up as text in Matrix. I must have missed something in the message format.

**Comment by remram44** [comment]
Apparently the sender is expected to render markdown into HTML and add the undocumented keys `"format": "org.matrix.custom.html", "formatted_body": "html_here"` to the message.

**Comment by remram44** [comment]
markdown2 seems like a good choice.

```
markdown2.Markdown(
    extras=['code-friendly', 'fenced-code-blocks', 'footnotes', 'link-patterns', 'tables'],
    link_patterns=link_patterns).convert(...)
```

**Comment by remram44** [comment]
Implemented by ad665119.

**Comment by remram44** [comment]
Looks like markdown2 has some serious bug and does correctly format GitHub Flavored Markdown: trentm/python-markdown2#226

**Comment by remram44** [comment]
New option: [Markdown](https://github.com/waylan/Python-Markdown) with [pymdown-extensions](https://github.com/facelessuser/pymdown-extensions).

``` python
import markdown
markdown.markdown('...', ['pymdownx.github'])
```

**Comment by remram44** [comment]
Implemented by 644039f3.
